test/subscribers/publish_zync_event_subscriber_test.rb

```diff
@@ -5,12 +5,20 @@ def setup
     @subscriber = PublishZyncEventSubscriber.new
   end
 
-  def test_create
-    application = FactoryBot.build_stubbed(:simple_cinstance, tenant_id: 1)
-    event = Applications::ApplicationCreatedEvent.new(application: application)
+  test 'publish Zync Event for OIDC auth' do
+    service = FactoryBot.create(:simple_service, backend_version: 'oauth')
+    application = FactoryBot.create(:simple_cinstance, service: service, tenant_id: 1)
+    event = Applications::ApplicationCreatedEvent.create(application, nil)
     assert @subscriber.call(event)
   end
 
+  test 'do not publish Zync Event for non-OIDC auth' do
+    service = FactoryBot.create(:simple_service)
+    application = FactoryBot.create(:simple_cinstance, service: service, tenant_id: 1)
```

Comment:
A little weird that you have to set `tenent_id`. Is this not set automatically?

Reply:
Maybe triggers are not loaded for testing? I usually have to run `rake multitenant:test:triggers` to create them.